src/actions/unstake.ts

```diff
@@ -16,19 +16,30 @@ export const unstake = async (
     throw Error("You can only unstake a non-zero amount of tokens");
 
   const depositsLength = await corePool.getDepositsLength(address);
-  if (depositsLength.eq(ethers.BigNumber.from(0)))
+
+  if (depositsLength.eq(ethers.BigNumber.from(0))) {
     throw Error("There are no deposits for you to unstake");
+  }
+
+  // Given depositId cannot be greater than the zero-based number of deposits
+  if (ethers.BigNumber.from(depositId).gte(depositsLength)) {
```

Comment:
Can you lend some context to what `depositId` and `depositsLength` are and what this check represents? I'm a bit unfamiliar with zFI.

Reply:
If you stake in a pool you create a `deposit`, and when you claim your rewards from that stake they are automatically staked and locked for the defined period, we use a year. This creates a second `deposit`, so the `depositId` is a unique identifier for a user's deposits over time. 

This check ensures that you aren't asking for a deposit that is within the number of deposits a user has made in the staking pool.

If you ask for `depositId = 10000` but there are only say `4` deposits, this will fail. And I'm actually realizing now that because it's 0-based we need `gt` not `gte`

Reply:
Awesome. Thanks for clarifying this. Glad to be rubber duck 🦆